Kernel/Ordering.cpp

```diff
@@ -289,6 +272,35 @@ Ordering::Result PrecedenceOrdering::compare(Literal* l1, Literal* l2) const
   return comparePredicates(l1, l2);
 } // PrecedenceOrdering::compare()
 
+bool Ordering::containsVar(AppliedTerm s, TermList var)
+{
+  ASS(var.isVar());
+  if (!s.aboveVar) {
+    return s.term.containsSubterm(var);
+  }
+  if (s.term.isVar()) {
+    return (*s.applicator)(s.term.var()).containsSubterm(var);
+  }
+  VariableIterator vit(s.term.term());
+  while (vit.hasNext()) {
+    auto v = vit.next();
+    if ((*s.applicator)(v.var()).containsSubterm(var)) {
+      return true;
+    }
+  }
+  return false;
+}
```

Comment:
Is there a reason why this is an instance function of `Ordering` and not of `AppliedTerm`? Feels more intuitive to me, but maybe I'm missing something.

Reply:
Good point, I added this here because `KBO` and `LPO` both use it but now that you mention it that would be more intuitive.